apps/rush-lib/src/schemas/rush.schema.json

```diff
@@ -211,8 +211,19 @@
       "type": "object",
       "properties": {
         "url": {
-          "description": "The remote url of the repository. If a value is provided, \"rush change\" will use it to find the right remote to compare against.",
-          "type": "string"
+          "oneOf": [
+            {
+              "description": "The remote url of the repository. If a value is provided, \"rush change\" will use it to find the right remote to compare against.",
+              "type": "string"
+            },
+            {
+              "description": "All allowed remote urls of the repository. If a value is provided, \"rush change\" will use one of these to find the right remote to compare against. Specifying multiple URLs is useful if a GitHub repository is renamed or for \"<projectName>.visualstudio.com\" vs \"dev.azure.com/<projectName>\" URLs.",
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            }
+          ]
```

Comment:
Polymorphic datastructures structures are counterintuitive, as a general rule.  For example if a person is editing a config file and all they see is `"url": "http://example"`, it's difficult to guess that multiple values are possible.  It  complicates code that processes these inputs, for example the JSON schema error messages may be less straightforward because they have to mention the `"oneOf"` alternatives.  (Also  `"url": ["http://one", "http://two"]` simply looks odd, as it deviates from our standard of pluralizing the name of an array.)

It might be better introduce a separate field "urls" and (ideally) deprecate the old field "url".